README.md

```diff
@@ -1,45 +1,140 @@
 # PostCSS Customprop Validate
 
-[PostCSS] plugin to validate default values of CSS custom properties.
+[PostCSS] plugin to validate [fallback values] of CSS custom properties.
 
-[PostCSS]: https://github.com/postcss/postcss
+Validating the fallback values of CSS custom properties manually can be error prone and challening, specially if there are many.
 
-```css
-.foo {
-    /* Input example */
-}
-```
+This plugin validates the CSS and returns custom properties with incorrect fallback values. It can also modify the source CSS file with the updated values.
 
-```css
-.foo {
-  /* Output example */
-}
+## Usage
+
+```js
+const { readFileSync } = require("fs");
+const postcss = require("postcss");
+const path = require("path");
+
+const plugin = require("./index");
+
+const cssPath = path.resolve("./sample.css");
+const css = readFileSync(cssPath, "utf-8");
+const properties = {
+  /* key and expected fallback values of custom properties */
+};
+
+postcss([plugin({ properties, write: false, callback: () => {} })])
+  .process(css, { from: cssPath })
+  .then((result) => {
+    /* handle the result */
+  });
 ```
 
-## Usage
+See [PostCSS] docs for examples for your environment.
+
+## Options
+
+The plugin accepts an object containing 3 properties:
+
+### **`properties`**
 
-**Step 1:** Install plugin:
+Required: `true`
 
-```sh
-npm install --save-dev postcss postcss-customprop-validate
+An object containg key and expected fallback values of custom properties. The plugin uses it as the source of truth to validate the CSS.
+
+Example:
+
+```json
+{
+  "--border-radius": "4px",
+  "--color-red": "#ff0000",
+  "--danger-border":
+    "var(--border-radius, 4px) solid var(--color-red, #ff0000)",
+};
 ```
 
-**Step 2:** Check you project for existed PostCSS config: `postcss.config.js`
-in the project root, `"postcss"` section in `package.json`
-or `postcss` in bundle config.
+### **`write`**
+
+Default: `false`
+
+Modifies the source CSS file with the correct fallback values.
+
+#### Formatting
 
-If you do not use PostCSS, add it according to [official docs]
-and set this plugin in settings.
+The plugin does not preserves the code formatting. If you use a code formatter like [prettier], you may need to re-run it after the plugin modifies the source file.
 
-**Step 3:** Add the plugin to plugins list:
+### **`callback`**
```

Comment:
@abhinavpreetu can you add here when is the callback called?

Reply:
The callback is called every time the plugin runs. Should I mention this?

Reply:
If there are multiple files, then after processing each file. Or if there is a single file or CSS string, after processing the CSS